cmd/gpu_plugin/gpu_plugin_test.go

```diff
@@ -51,13 +61,7 @@ func TestScan(t *testing.T) {
 	sysfs := path.Join(root, "sys")
 	devfs := path.Join(root, "dev")
 
-	tcases := []struct {
-		name         string
-		devfsdirs    []string
-		sysfsdirs    []string
-		sysfsfiles   map[string][]byte
-		expectedDevs int
-	}{
+	tcases := []testCase{
```

Comment:
If we add one more table-based test we'll need a corresponding type for cases as they are test specific usually. It may be easier to look up the type when it's defined in place while adding new test cases.

Reply:
Do you mean that I should move type definition after Notify() so that is just before TestScan()?  Or something else?

Reply:
My recommendation is not to introduce the `testCase` type at all. It doesn't add up to readability.

Reply:
It's not there for readability, but for scanWithParams().  All function parameters need a type!

Reply:
@rojkov Is there still something unclear about this, or can it be resolved?

Reply:
I agree with @rojkov here. Adding new field 'shareCounts []int' to the test case and getting rid of scanWithParams would look better in my opinion.

Reply:
@bart0sh Command line options are another *dimension* to testing, so at least they would need to be separate array (there will be more command line options, and trying to flatten multi-dimensional test-array to single dimension would just make it unmaintainable).

While having 4 loops inside each other + some conditional code in innermost 2 loops would be kind of OK for now, I think it will need to be lifted out to a separate function before rest of my changes are there.  I think it's better to do that early.

Result of all the changes would look something like this: https://github.com/eero-t/intel-device-plugins-for-kubernetes/blob/gpu-plugin-split-vfs/cmd/gpu_plugin/gpu_plugin_test.go

Reply:
I still don't see much benefits of having scanWithParams. However, I agree with @rojkov that the code is less readable because of introducing testCase type and moving piece of code into scanWithParams.
So, my proposal stays the same: get rid of scanWithParams and testCase.